### PR TITLE
Validate remote source endpoint and make permissions consistent

### PR DESCRIPTION
**Pull request by umbreak**
Fixes https://github.com/BlueBrain/nexus/issues/2212
Fixes https://github.com/BlueBrain/nexus/issues/2214
Fixes https://github.com/BlueBrain/nexus/issues/2189

### CODE REVIEW

**Comment by umbreak** [comment]
Fixes https://github.com/BlueBrain/nexus/issues/2212
Fixes https://github.com/BlueBrain/nexus/issues/2214

File: delta/plugins/composite-views/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/compositeviews/client/DeltaClient.scala [hunk 1]
```diff
@@ -48,8 +48,19 @@ final class DeltaClient(client: HttpClient, retryDelay: FiniteDuration)(implicit
...
+    * Checks whether the events endpoint and token provided by the source are correct
+    *
+    * @param source the source
+    */
+  def checkEvents(source: RemoteProjectSource): HttpResult[Unit] = {
+    val uri                              =
+      source.endpoint / "resources" / source.project.organization.value / source.project.project.value / "events"
+    implicit val cred: Option[AuthToken] = token(source)
+    client(Head(uri).withCredentials) {
+      case resp if resp.status.isSuccess() => UIO.delay(resp.discardEntityBytes()) >> IO.unit
```
Comment (bogdanromanx) [inline delta/plugins/composite-views/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/compositeviews/client/DeltaClient.scala:61]: and if it's not successful?

File: delta/plugins/composite-views/src/test/scala/ch/epfl/bluebrain/nexus/delta/plugins/compositeviews/indexing/CompositeIndexingSpec.scala [hunk 2]
```diff
@@ -93,7 +93,7 @@ class CompositeIndexingSpec
   implicit private val caller: Caller   = Caller(bob, Set(bob, Group("mygroup", realm), Authenticated(realm)))
   implicit private val baseUri: BaseUri = BaseUri("http://localhost", Label.unsafe("v1"))
 
-  private val allowedPerms = Set(permissions.query, resources.read)
+  private val allowedPerms = Set(permissions.query, events.read)
```
Comment (bogdanromanx) [inline delta/plugins/composite-views/src/test/scala/ch/epfl/bluebrain/nexus/delta/plugins/compositeviews/indexing/CompositeIndexingSpec.scala:96]: This also needs to be added to the API changes doc.

**Review by bogdanromanx — CHANGES_REQUESTED** [review]

File: delta/plugins/composite-views/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/compositeviews/client/DeltaClient.scala (hunk 1 above)
Comment (umbreak) [reply delta/plugins/composite-views/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/compositeviews/client/DeltaClient.scala:61]: The it with go on the left raiseerror channel of it IO

Comment (bogdanromanx) [reply delta/plugins/composite-views/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/compositeviews/client/DeltaClient.scala:61]: will the entity bytes be discarded?

Comment (umbreak) [reply delta/plugins/composite-views/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/compositeviews/client/DeltaClient.scala:61]: Yes, [that's the case](https://github.com/BlueBrain/nexus/blob/master/delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/http/HttpClient.scala#L101).

**Review by bogdanromanx — APPROVED** [review]

**Pull request merged**